pkg/security/security_test.go

```diff
@@ -62,60 +103,31 @@ func (s *testSecuritySuite) TestToTLSConfig(c *C) {
 		SSLKey:  filepath.Join(temp, "ssl.key"),
 	}
 
-	// These certs are generated with:
-	//
-	// ```sh
-	// # generate CA keys
-	// openssl ecparam -name secp224r1 -genkey -noout -out ca.key
-	// openssl req -x509 -new -nodes -key ca.key -days 999999 -out ca.crt -subj '/CN=localhost'
-	//
-	// # generate SSL keys
-	// openssl ecparam -name secp224r1 -genkey -noout -out ssl.key
-	// openssl req -new -key ssl.key -out ssl.csr -subj '/CN=localhost'
-	// openssl x509 -req -in ssl.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out ssl.crt -days 999999
-	// ```
-
-	err := ioutil.WriteFile(dummyConfig.SSLCA, []byte(`
------BEGIN CERTIFICATE-----
-MIIBBjCBtQIJAMLMVjQw2v1pMAoGCCqGSM49BAMCMBQxEjAQBgNVBAMMCWxvY2Fs
-aG9zdDAgFw0xOTA0MTcxODEyNDNaGA80NzU3MDMxMzE4MTI0M1owFDESMBAGA1UE
-AwwJbG9jYWxob3N0ME4wEAYHKoZIzj0CAQYFK4EEACEDOgAELwEHdmAcDtBYK9BH
-72q0dKbBBqIG7MZ5+qc+LTcz0OSdhuWkWUZkNN6MqKAPuP7nSo1+21Vb8YswCgYI
-KoZIzj0EAwIDQAAwPQIcbNvV16rOOzwotH65cJY6cCdf0h3IODjlWMf1qAIdAIBB
-Fma6g8iW5zdQPqDR9BGqugNPjtI/SMK6tfQ=
------END CERTIFICATE-----
-	`), 0644)
+	err := ioutil.WriteFile(dummyConfig.SSLCA, []byte(testCa), 0644)
 	c.Assert(err, IsNil)
 
-	err = ioutil.WriteFile(dummyConfig.SSLCert, []byte(`
------BEGIN CERTIFICATE-----
-MIIBBTCBtAIJAP8wfS+6tJ3LMAkGByqGSM49BAEwFDESMBAGA1UEAwwJbG9jYWxo
-b3N0MCAXDTE5MDQxNzE4MTI0NFoYDzQ3NTcwMzEzMTgxMjQ0WjAUMRIwEAYDVQQD
-DAlsb2NhbGhvc3QwTjAQBgcqhkjOPQIBBgUrgQQAIQM6AAQJaXEnDhG2tPxD4wl1
-ycaZwqWm9JeQZFuUPgxekGwCMM22sKpYLvhdKroSBoKWwXIC6vZMWeIj/zAJBgcq
-hkjOPQQBA0EAMD4CHQC05dXi9zFLjYjQGhpJNx+Nc/5vC6E7j/MU+xsTAh0A6SUn
-g916djuFWv8djdDq+0NEFD9OzgPdSb8rZw==
------END CERTIFICATE-----
-	`), 0644)
+	err = ioutil.WriteFile(dummyConfig.SSLCert, []byte(testCert), 0644)
 	c.Assert(err, IsNil)
 
-	err = ioutil.WriteFile(dummyConfig.SSLKey, []byte(`
------BEGIN EC PRIVATE KEY-----
-MGgCAQEEHCsPBVueZ3YX3yp1tn15YXj0cTKGCo1SO1EWO92gBwYFK4EEACGhPAM6
-AAQJaXEnDhG2tPxD4wl1ycaZwqWm9JeQZFuUPgxekGwCMM22sKpYLvhdKroSBoKW
-wXIC6vZMWeIj/w==
------END EC PRIVATE KEY-----
-	`), 0600)
+	err = ioutil.WriteFile(dummyConfig.SSLKey, []byte(testKey), 0600)
 	c.Assert(err, IsNil)
 
 	config, err := dummyConfig.ToTLSConfig()
 	c.Assert(err, IsNil)
 	c.Assert(config, NotNil)
 	c.Assert(config.RootCAs.Subjects(), HasLen, 1)
-	c.Assert(config.Certificates, HasLen, 1)
-	sslKey, ok := config.Certificates[0].PrivateKey.(*ecdsa.PrivateKey)
-	c.Assert(ok, IsTrue)
-	c.Assert(sslKey.Curve, Equals, elliptic.P224())
+	for i := 0; i < 2; i++ {
+		var cert *tls.Certificate
+		if i == 0 {
+			cert, err = config.GetCertificate(nil)
+		} else {
+			cert, err = config.GetClientCertificate(nil)
+		}
+		c.Assert(err, IsNil)
+		sslKey, ok := cert.PrivateKey.(*ecdsa.PrivateKey)
+		c.Assert(ok, IsTrue)
+		c.Assert(sslKey.Curve, Equals, elliptic.P224())
+	}
```

Comment:
This looks clearer if you just expand the loop.

Reply:
done in https://github.com/pingcap/tidb-binlog/pull/927/commits/67e744ed8221586388287667b990338cdd727abb